src/librustdoc/html/render.rs

```diff
@@ -4266,10 +4264,7 @@ fn item_foreign_type(w: &mut fmt::Formatter<'_>, cx: &Context, it: &clean::Item)
     render_assoc_items(w, cx, it, it.def_id, AssocItemRender::All)
 }
 
-impl Print for Sidebar<'_> {
-    fn print(self, buffer: &mut Buffer) {
-        let cx = self.cx;
-        let it = self.item;
+    fn print_sidebar(cx: &Context, it: &clean::Item, buffer: &mut Buffer) {
```

Comment:
I could, but I think it's not too likely we want to do that in this PR since there are existing functions where the fmt::Formatter was in the middle of the function or so and moving that around is, while not generally error-prone due to types, quite noisy. I'd like to avoid doing it in this PR :)